src/test/java/org/opensearch/security/transport/SecurityRequestHandlerTests.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.transport;
+
+import org.junit.Test;
+import org.opensearch.extensions.ExtensionScopedSettings;
+import org.opensearch.extensions.ExtensionsSettings;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SecurityRequestHandlerTests {
+
+    private SecurityRequestHandler securityRequestHandler = new SecurityRequestHandler(
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null
+    );
+
+    @Test
+    public void testShouldAllowExtensionRequest() {
+        String principal = "CN=transport-0.example.com, OU=SSL, O=Test, L=Test, C=DE";
+        ExtensionsSettings.Extension extension = new ExtensionsSettings.Extension(
+            "name",
+            "uniqueId",
+            "hostAddress",
+            "port",
+            "version",
+            "opensearchVersion",
+            "minimumCompatibleVersion",
+            List.of("CN=transport-1.example.com, OU=SSL, O=Test, L=Test, C=DE", "CN=transport-0.example.com, OU=SSL, O=Test, L=Test, C=DE"),
```

Comment:
These should be passed in the `additionalSettings`. With the `ExtensionAwarePlugin` extension point, it allows a plugin (the security plugin) to define settings that can be passed in the extension initialization request and those will get added to the `ExtensionsSettings.Extension` as additional settings.